core/include/detray/propagator/propagator.hpp

```diff
@@ -134,6 +134,9 @@ struct propagator {
         // Run all registered actors/aborters after init
         run_actors(actor_states, propagation);
 
+        // Find next candidate
+        propagation._heartbeat &= _navigator.update(propagation);
```

Comment:
We also need to call the update function in case the track is on the surface at the beginning